zephyr/wrapper.c

```diff
@@ -724,11 +730,12 @@ int arch_cpu_secondary_cores_prepare_d0ix(void)
 void arch_cpu_disable_core(int id)
 {
 	/* TODO: call Zephyr API */
+	w_core_enable_mask &= ~BIT(id);
 }
 
 int arch_cpu_is_core_enabled(int id)
 {
-	return arch_cpu_active(id);
+	return BIT(id) & w_core_enable_mask;
```

Comment:
nitpick: schematically I would check if the `BIT(id)` is set in `w_core_enable_mask`:
```c
return w_core_enable_mask & BIT(id);
```
Which is how for example  the implementation in `src/arch/xtensa/lib/cpu.c` does.

Reply:
Updated.